go-controller/pkg/ovn/egressip.go

```
@@ -1785,7 +1789,7 @@ func (e *egressIPController) handleEgressReroutePolicy(podIPNets []*net.IPNet, s
 	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
 	gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
 	if err != nil {
-		return fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
+		return fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
```

Why is the change on `getGatewayRouterJoinIP` not being cherry-picked?

---

same here: https://github.com/openshift/ovn-kubernetes/blob/feab42ac6bf0d4613159120b65ba9fe24cf980d2/go-controller/pkg/ovn/egressip.go#L1775 its not really returning the error in 4.10, only in 4.11 Mohamed did https://github.com/ovn-org/ovn-kubernetes/pull/2965/files#diff-0f91bbc7bdb15165a9fc2e1cf7492425cbf1cbd35bf3daa22a000fc4027c9b9eR1922 which needed to be fixed until 4.11

---

I mean that `%v` was replaced by `%w`, seems a bit arbitrary not cherry-picking that, but maybe there is a reason.

---

The same with `handleEgressReroutePolicy`.

---

I had no idea what `%w` was doing actually.

---

Its a trick patryk taught me in golang. So for us to use the `errors.Is(err, libovsdbclient.ErrNotFound)` we should return the error as %w to be able to parse the type of error correctly, parsing %v format doesn't yield the expected result.

---

honestly looking back into this code now as I rebase... I feel we should just cherry-pick that part from mohamed's fix here: https://github.com/ovn-org/ovn-kubernetes/pull/2965/files#diff-0f91bbc7bdb15165a9fc2e1cf7492425cbf1cbd35bf3daa22a000fc4027c9b9eR1922 because it makes no sense that on error we log it and proceed to use that to doing `MatchIPNetFamily` ... my tests also fail because of the flawed logic. I am updating this part to fix this from getting the changes in `getGatewayRouterJoinIP`.